### PR TITLE
Landing page '/'

### PR DESCRIPTION
**Pull request by alaa-sahloub**
Relates #29

### Create index.js file inside HomePage folder in the Pages folder which contains:

- Import React and useState and useEffect from react.
- import Axios to make a request.
- import message from ant design.
- import Search and CardContainer component.
- import styles.css file.
- create HomePage function:

 > create data use state and initialize it as an array.
 > create fetchData function which use axios to make a request to get data from /api/v1/kindergarten endpoint.
 > return the data.
 > catch the error in the message form ant design.

- Invoke the fetchData function in the useEffect.
- render header image and the title.
- render Search and CardContainer components.
- export HomePage function.

### CODE REVIEW

File: client/src/Components/CardContainer/index.js [hunk 1]
```diff
@@ -9,15 +9,24 @@ const { Title } = Typography;
...
         {data.length !== 0 ? (
           data.map((item) => (
             <li key={item.id}>
-              <Card {...item} />
+              <Card
+                id={item.id}
+                title={item.kindergarten_name}
+                rating={parseFloat(item.rating_average)}
+                reviewersNo={parseFloat(item.rating_count)}
+                location={item.location_sub}
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:25]:
```suggestion
                mainLocation={item.location_main}
                subLocation={item.location_sub}
```
And at the `Card` component edit these lines 
```js
const CardComponent = ({
  id,
  title,
  rating,
  reviewersNo,
  subLocation,
  mainLocation,
  minPrice,
  maxPrice,
  cover,
}) => {
```

```jsx
<h3> {`${mainLocation} - ${subLocation}`}</h3>
```

```js
CardComponent.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  reviewersNo: PropTypes.number.isRequired,
  subLocation: PropTypes.string.isRequired,
  mainLocation: PropTypes.string.isRequired,
  minPrice: PropTypes.number.isRequired,
  maxPrice: PropTypes.number.isRequired,
  cover: PropTypes.string.isRequired,
};
```

File: client/src/Pages/HomePage/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          if (a.rating_average > b.rating_average) {
+            return -1;
+          }
+          return 0;
+        });
+
+        setData(sortRes);
+      })
+      .catch(() => message.err('خطأ في السيرفر يرجى المحاولة لاحقًا.'));
+  };
```
Comment (Mu7ammadAbed) [inline client/src/Pages/HomePage/index.js:29]:
```suggestion
  const [kindergartensData, setKindergartensData] = useState([]);

  const fetchData = async () => {
    try {
      const {
        data: { data },
      } = await axios.get(`/api/v1/kindergarten`);
      const sortRes = data.sort((a, b) => {
        if (a.rating_average < b.rating_average) {
          return 1;
        }
        if (a.rating_average > b.rating_average) {
          return -1;
        }
        return 0;
      });
      setKindergartensData(sortRes);
    } catch (err) {
      message.error('خطأ في السيرفر يرجى المحاولة لاحقًا.');
    }
  };
```

File: client/src/Pages/HomePage/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <div>
+      <div className="header-image">
+        <p className="header-image-title">
+          المنصة الأولى لمساعدتكم في اختيار روضة أبنائكم
+        </p>
+      </div>
+      <div className="imported-component-container">
+        <Search dorpListOptions={[{ id: 2, value: 'asda', disabled: false }]} />
+        <CardContainer data={data} />
```
Comment (Mu7ammadAbed) [inline client/src/Pages/HomePage/index.js:43]:
```suggestion
        <CardContainer data={kindergartensData} />
```

File: client/src/Pages/HomePage/style.css [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  position: absolute;
+  left: 231px;
+  top: 185px;
+  font-style: normal;
+  font-weight: bold;
+  font-size: 45px;
+  line-height: 38px;
+  color: #fdf6f6;
+  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+}
```
Comment (Mu7ammadAbed) [inline client/src/Pages/HomePage/style.css:24]: new line here

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: client/src/Components/CardContainer/index.js [hunk 5]
```diff
@@ -49,7 +51,8 @@ CardContainer.propTypes = {
       title: PropTypes.string,
       rating: PropTypes.number,
       reviewersNo: PropTypes.number,
-      location: PropTypes.string,
+      subLocation: PropTypes.string.isRequired,
+      mainLocation: PropTypes.string.isRequired,
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:55]:
```suggestion
      subLocation: PropTypes.string,
      mainLocation: PropTypes.string,
```

**Review by Mu7ammadAbed — DISMISSED** [review]

File: client/src/Pages/HomePage/index.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      });
+      setKindergartensData(sortRes);
+    } catch (err) {
+      message.error('خطأ في السيرفر يرجى المحاولة لاحقًا.');
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
```
Comment (Osama-you) [inline client/src/Pages/HomePage/index.js:33]: when you make  REST call in useEffects you have to cleanup it
that link will help you to do that:
https://stackoverflow.com/questions/53861916/canceling-an-axios-rest-call-in-react-hooks-useeffects-cleanup-failing

File: client/src/Components/CardContainer/index.js [hunk 7]
```diff
@@ -9,15 +9,26 @@ const { Title } = Typography;
 
 const CardContainer = ({ data, searchText }) => (
   <div>
```
Comment (Mu7ammadAbed) [inline client/src/Components/CardContainer/index.js:11]:
```suggestion
  <>
```

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Osama-you — APPROVED** [review]

**Review by Israa91 — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Pull request merged**